content/articles/react-drf-shortener/index.md

```diff
@@ -0,0 +1,369 @@
+-   Understanding of the Fetch API.
+-   Python and Node.js installed on the computer.
+-   A browser, preferably Google Chrome.
+-   A code editor, preferably VS Code.
+
+### Goals
+
+By the end of this article, the reader should be able to:
+
+-   Create a custom URL shortener.
+-   Work with React and Django.
+-   Fetch data from a database using Django and display it on React webpage.
+
+### Setting up the Backend and Exposing the API Routes.
+
+Create a folder named `react-drf-shortener` and `cd` into it. First, create a virtual environment for the Django backend. More information about Django virtual environments can be found [here](https://docs.python.org/3/tutorial/venv.html).
+
+Run the commands below in the terminal to create the virtual environment, activate it, and install Django in it:
+
+```bash
+python -m venv shortener-env
+
+.\shortener-env\Scripts\activate
+
+python -m pip install Django
+```
+
+After Django has been successfully installed, run the following commands in the same directory to create a Django project named `urlshortener` and open it in VS Code:
+
+```bash
+django-admin startproject urlshortener
+
+code urlshortener
+```
+
+Open the integrated terminal in the code editor and ensure that the virtual environment is still active. If it is not, run the following commands to activate it:
+
+```bash
+cd ..
+
+.\shortener-env\Scripts\activate
+
+cd urlshortener
+```
+
+The above commands change our active directory to the root directory, activate the virtual environment, and then navigate to the `urlshortener` folder.
+
+Run the following code to make sure that the server is working:
+
+```bash
+python manage.py runserver
+```
+
+If the server is working perfectly, this will open a development server at `http://127.0.0.1:8000/`
+
+Now that the server is working perfectly, we will create a new app to handle all the API requests from the front end. Run the code below:
+
+```bash
+django-admin startapp api
+```
+
+Next, run the commands below to install all the packages/dependencies we will need:
+
+```bash
+pip install djangorestframework
+
+python -m pip install django-cors-headers
+```
+
+**Code Explanation**:
+
+`pip install djangorestframework` will install Django Rest Framework(DRF). DRF allows us to create an API using Django.
+
+`python -m pip install django-cors-headers` - This will install CORS to allow the Django backend to communicate with the React front-end.
+
+### Registering the api app and the installed packages
+
+Open the `settings.py` file and modify its contents as shown below:
+
+For the `INSTALLED_APPS` section:
+
+```python
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+
+    'api.apps.ApiConfig', #registers our api app
+
+    'rest_framework',
+
+    'corsheaders'
+]
+```
+
+For the `MIDDLEWARE` section:
```

Comment:
What is it for? Please add 1 - 2 lines explanation about middlewares and each item of the below list.